### PR TITLE
3ds: add wslay and nettle

### PR DESCRIPTION
**Pull request by Pixel-Pop**
Here is a working example program, adapted from wslay's examples, with the 3ds connecting to a websocat server: [wslay-client.zip](https://github.com/devkitPro/pacman-packages/files/6673296/wslay-client.zip)

`gcc` is in nettle's `makedepends` because it seems nettle needs to build utilities to run on the build system at build time.

Please feel free to make any changes.

### CODE REVIEW

File: 3ds/nettle/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Maintainer:  Dave Murphy <davem@devkitpro.org>
+
+pkgname=3ds-nettle
```
Comment (WinterMute) [inline 3ds/nettle/PKGBUILD:3]: Just remove the nettle package, it's not necessary for the wslay library. It's only used for examples which won't be built if nettle isn't installed.

File: 3ds/wslay/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+pkgname=3ds-wslay
+pkgver=1.1.1
+pkgrel=1
+pkgdesc='Wslay is a WebSocket library written in C.'
+arch=('any')
+url='https://github.com/tatsuhiro-t/wslay'
+license=('MIT')
+options=(!strip libtool staticlibs)
+groups=('3ds-portlibs')
+makedepends=('3ds-pkg-config' 'devkitpro-pkgbuild-helpers')
```
Comment (WinterMute) [inline 3ds/wslay/PKGBUILD:12]: devkitpro-pkgbuild-helpers has been superceded by dkp-toolchain-vars and several other packages that aren't needed for this library. Just use dkp-toolchain-vars here.

**Review by WinterMute — CHANGES_REQUESTED** [review]

**Comment by WinterMute** [comment]
Don't worry about merging upstream into your fork when doing PRs, this can end up causing conflicts. If nothing you're editing has changed then you can just ignore it, otherwise it's better to rebase from upstream instead.

**Comment by Pixel-Pop** [comment]
Thank you for your feedback. Should I open a separate PR for nettle?

**Review by WinterMute — APPROVED** [review]

**Comment by WinterMute** [comment]
> Thank you for your feedback. Should I open a separate PR for nettle?

Thanks for making the changes so quickly. I've merged and packaged this now. 

I'm not sure nettle is really a good fit for 3DS unless there's something specific you're using it for that can't be done with mbedtls for instance. If nettle is added though then this package will need modding to avoid building examples.

**Pull request merged**